src/base/bittorrent/torrentinfo.cpp

```diff
@@ -76,13 +77,45 @@ TorrentInfo TorrentInfo::load(const QByteArray &data, QString *error) noexcept
 
 TorrentInfo TorrentInfo::loadFromFile(const QString &path, QString *error) noexcept
 {
+    if (error)
+        error->clear();
+
+    QFile file {path};
+    if (!file.open(QIODevice::ReadOnly)) {
+        if (error)
+            *error = QCoreApplication::translate("TorrentInfo", "File open error");
+        return TorrentInfo();
+    }
+
+    const qint64 fileSizeLimit = 100 * 1024 * 1024;  // 100 MB
+    if (file.size() > fileSizeLimit) {
+        if (error)
+            *error = QCoreApplication::translate("TorrentInfo", "File size exceeds max limit %1").arg(fileSizeLimit);
+        return TorrentInfo();
+    }
+
+    const QByteArray data = file.read(fileSizeLimit);
```

Comment:
As I said before, `read()` can also fail.

Reply:
Added a isEmpty check, however there is no way to get the true error as stated in qt doc.

Reply:
I think the valid condition of error is "data.size() != file.size()". But I personally would use other read() overload (taken buffer as param).

Reply:
>I think the valid condition of error is "data.size() != file.size()". 

alright.

>As I said before, read() can also fail.

Just curious, in what condition? the file is opened for reading only and the file pointer is unmodified at start.

Reply:
Disk error?..